src/neural/encoder.cc

```diff
@@ -35,17 +35,136 @@ namespace {
 const int kMoveHistory = 8;
 const int kPlanesPerBoard = 13;
 const int kAuxPlaneBase = kPlanesPerBoard * kMoveHistory;
+
+int CompareTransposing(uint64_t value, int initial_transform) {
+  if ((initial_transform & 1) != 0) {
+    value = ReverseBitsInBytes(value);
+  }
+  if ((initial_transform & 2) != 0) {
+    value = ReverseBytesInBytes(value);
+  }
+  auto alternative = TransposeBitsInBytes(value);
+  if (value < alternative) return -1;
+  if (value > alternative) return 1;
+  return 0;
+}
+
+int ChooseTransform(const ChessBoard& board) {
```

Comment:
It's quite a heavy function, would be interesting to see the performance hit, especially near endgames.

Reply:
Its never called unless input format type is 3.
Its actually not too expensive unless the side to move king is on the diagonal with symmetrical other pieces, but even then, this function is not called within PickNodeToExtend inner loop nor the other significantly expensive loops.  Its only called once per node.  In that context it is not very heavy even if it was called outside of input_type 3.
Actually performing the transforms on all 112 bit planes is going to be considerably more expensive (but still not so expensive as to be a concern).